src/formatDate.js

```diff
@@ -8,7 +8,29 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateParts = date.split(fromFormat[3]);
+  const newDate = [];
+  const dateSet = {};
+
+  for (let i = 0; i < 3; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  if (dateSet.YYYY) {
+    dateSet.YY = dateSet.YYYY.slice(2);
+  }
+
+  if (dateSet.YY < 30) {
+    dateSet.YYYY = '20' + `${dateSet.YY}`;
+  } else {
+    dateSet.YYYY = '19' + `${dateSet.YY}`;
+  }
+
+  for (let i = 0; i < 3; i++) {
+    newDate.push(dateSet[toFormat[i]]);
+  }
+
+  return newDate.join(toFormat[toFormat.length - 1]);
```

Comment:
The function does not handle the case where the 'fromFormat' or 'toFormat' might have different separators. You need to ensure that the correct separator is used when splitting and joining the date parts.